openshift/machine-config-operator
# OCPBUGS-8446: MCO-503: daemon: have a special path to sync in certs

Have a new routine that watches for controllerconfig changes to read the latest kubelet ca bundle, and lay that down to disk directly. File writing and validation will skip over it.

This is still WIP but the functionality should work. You can try this by pausing the pools and forcing a rotation of any of the certificates in the bundle. Although it won't be instant, the MCD will sync it once the safety controllerconfig sync happens.

I tried doing this via an additional configmap but it feels like this is a bit cleaner, and allows the MCD to have limited permissions in what it can read.

- What I did

1. Add a new controllerconfig watcher + sync to directly write the newest kubelet ca bundle directly to disk
2. Remove MCD file writing and verification for that file
3. Add a test to write the file
4. Remove (now incorrect) cert metric and alert
5. Add additional path in the MCS to serve the latest bundle as well

- How to verify it

1. Install a 4.13 cluster without this change and pause pools, in ~24 hours you will get kubeletdown critical alerts. With this change, you should no longer see it
2. Faster way to verify: pause pools, manually expire a cert (e.g. kube-apiserver-to-kubelet-signer), and see the change propagate to disk without unpause

Note that this is only a temporary 4.13 solution, with future steps outlined in https://issues.redhat.com/browse/MCO-499 for a more comprehensive solution
